From 62bab2a0d88153abada8aa3c6ea1659e41e7ff5d Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:00 +0800
Subject: [PATCH] sw64: switch GUP to the generic get_user_pages_fast()
 implementation
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 62bab2a0d88153abada8aa3c6ea1659e41e7ff5d
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56U83

--------------------------------

sw64 fits all conditions required by the generic get_user_pages_fast()
code, so activate this feature by default.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/sw_64/Kconfig b/arch/sw_64/Kconfig
index 347958e60023..6c9fb51347f3 100644
--- a/arch/sw_64/Kconfig
+++ b/arch/sw_64/Kconfig
@@ -12,6 +12,7 @@ config SW64
 	select HAVE_PCSPKR_PLATFORM
 	select HAVE_PERF_EVENTS
 	select HAVE_GENERIC_HARDIRQS
+	select HAVE_GENERIC_GUP
 	select GENERIC_CLOCKEVENTS
 	select GENERIC_IRQ_PROBE
 	select GENERIC_IRQ_LEGACY
-- 
2.34.1

